copier/tools.py

```diff
@@ -232,3 +233,24 @@ def escape_git_path(path: str) -> str:
         lambda match: "".join(f"\\{whitespace}" for whitespace in match.group()),
         path,
     )
+
+
+def with_git_object_dirs(git_cmd: LocalCommand, *object_dirs: Path) -> LocalCommand:
+    """Enhance a Git command to search for Git objects in additional directories.
+
+    See https://git-scm.com/docs/git#Documentation/git.txt-codeGITALTERNATEOBJECTDIRECTORIEScode
+    for more details.
+
+    Args:
+        git_cmd: The Git command.
+        *object_dirs: The Git object directories.
+
+    Returns:
+        LocalCommand: The Git command that searches for Git objects in the
+            specified Git object directories.
+    """
+    return git_cmd.with_env(
+        GIT_ALTERNATE_OBJECT_DIRECTORIES=(";" if OS == "windows" else ":").join(
+            str(object_dir / ".git" / "objects") for object_dir in object_dirs
```

Comment:
`.git` may be a  file with this kind of contents:
`gitdir: pathtogitdir`

if the repo has been initialized with : 
```sh 
$ git init --separate-git-dir ../pathtogitdir
```

So `(object_dir / ".git" / "objects")` don't work.

Reply:
Great catch, thanks a ton! :ok_hand: I'll work on supporting a separate Git directory.